ui/components/Policies/PolicyViolations/PolicyViolationDetails.tsx

```diff
@@ -0,0 +1,187 @@
+import React from "react";
+import ReactMarkdown from "react-markdown";
+import remarkGfm from "remark-gfm";
+
+import styled from "styled-components";
+import { useGetPolicyValidationDetails } from "../../../hooks/policyViolations";
+import { PolicyValidation } from "../../../lib/api/core/core.pb";
+import { Kind } from "../../../lib/api/core/types.pb";
+import { formatURL } from "../../../lib/nav";
+import { V2Routes } from "../../../lib/types";
+import Flex from "../../Flex";
+import Link from "../../Link";
+import Page from "../../Page";
+import Text from "../../Text";
+import Timestamp from "../../Timestamp";
+
+import { AppContext } from "../../../contexts/AppContext";
+import { FluxObject } from "../../../lib/objects";
+import Parameters from "../Utilis/Parameters";
+import Severity from "../Utilis/Severity";
+import HeaderRows, { Header } from "../Utilis/HeaderRows";
+
+const SectionWrapper = ({ tilte, children }) => {
+  return (
+    <Flex column wide gap="8" data-testid="occurrences">
+      <Text bold color="neutral30">
+        {tilte}
+      </Text>
+      {children}
+    </Flex>
+  );
+};
+
+interface IViolationDetailsProps {
+  violation: PolicyValidation;
+  kind: string;
+}
+const ViolationDetails = ({ violation, kind }: IViolationDetailsProps) => {
+  const { setDetailModal } = React.useContext(AppContext);
+  const {
+    clusterName,
+    severity,
+    createdAt,
+    category,
+    howToSolve,
+    description,
+    violatingEntity,
+    entity,
+    namespace,
+    occurrences,
+    name,
+    parameters,
+  } = violation || {};
+
+  const entityObject = new FluxObject({ payload: violatingEntity });
+  const headers: Header[] = [
+    {
+      rowkey: "Policy Name",
+      value: name,
+    },
+    {
+      rowkey: "Application",
+      children: (
+        <Link
+          to={formatURL(
+            Kind[kind] === Kind.Kustomization
+              ? V2Routes.Kustomization
+              : V2Routes.HelmRelease,
+            {
+              name: entity,
+              namespace: namespace,
+              clusterName,
+            }
+          )}
+        >
+          {namespace}/{entity}
+        </Link>
+      ),
+    },
+    {
+      rowkey: "Violation Time",
+      value: <Timestamp time={createdAt} />,
+    },
+    {
+      rowkey: "Severity",
+      children: <Severity severity={severity || ""} />,
+    },
+    {
+      rowkey: "Category",
+      value: category,
+    },
+    {
+      rowkey: "Violating Entity",
+      children: (
+        <Text
+          pointer
+          size="medium"
+          color="primary"
+          onClick={() => setDetailModal({ object: entityObject })}
+        >
+          {entityObject.namespace}/{entityObject.name}
+        </Text>
+      ),
+    },
+  ];
+
+  return (
+    <Flex wide tall column gap="32">
+      <HeaderRows headers={headers} />
```

Comment:
The difference relies on 
```
interface Header {
  rowkey: string;
  value?: any;
  children?: any;
  visible?: boolean;
}
```

with much controlling on `HeaderRow` beside it's not globally used and is cover the needs here at Violations & Policies in the upcoming pr

Reply:
I don't understand the difference - it just seems really complex for something we already have

Reply:
The main difference is that when you create the listOf headers you set the the visible directly instead of doing the check and push/pop to a list also the display of the Flex not a table which is meant for this page